vslm/vslm_object_manager.go

```diff
@@ -0,0 +1,237 @@
+	if err != nil {
+		return nil, err
+	}
+	return &res.Returnval, nil
+}
+
+func (t *VslmTask) QueryInfo(ctx context.Context) (*types.VslmTaskInfo, error) {
+	req := types.VslmQueryInfo{}
+	res, err := methods.VslmQueryInfo(ctx, t.c, &req)
+	if err != nil {
+		return nil, err
+	}
+	return &res.Returnval, nil
+}
+
+func (t *VslmTask) Cancel(ctx context.Context) error {
+	req := types.VslmCancelTask{}
+	_, err := methods.VslmCancelTask(ctx, t.c, &req)
+	return err
+}
+
+type VslmObjectManager struct {
+	vim.ManagedObjectReference
+	c *Client
+}
+
+// NewVslmObjectManager returns an ObjectManager referecing the vslm VcenterVStorageObjectManager endpoint.
+// This endpoint is always connected to vpxd and utilizes the global catalog to locate objects and does
+// not require a datastore.  To connect to the VStorageObjectManager on the host or in vpxd use the vslm.ObjectManager type.
+func NewVslmObjectManager(client *Client) *VslmObjectManager {
+	mref := client.ServiceContent.VStorageObjectManager
+
+	m := VslmObjectManager{
+		ManagedObjectReference: mref,
+		c:                      client,
+	}
+
+	return &m
+}
+
+func (m *VslmObjectManager) CreateDisk(ctx context.Context, spec vim.VslmCreateSpec) (*VslmTask, error) {
+	req := types.VslmCreateDisk_Task{
+		This: m.Reference(),
+		Spec: spec,
+	}
+
+	res, err := methods.VslmCreateDisk_Task(ctx, m.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewVslmTask(m.c, res.Returnval), nil
+}
+
+func (m *VslmObjectManager) CreateDiskFromSnapshot(ctx context.Context, id vim.ID, snapshotId vim.ID, name string,
+	profile []vim.VirtualMachineProfileSpec, crypto *vim.CryptoSpec,
+	path string) (*VslmTask, error) {
+	req := types.VslmCreateDiskFromSnapshot_Task{
+		This:       m.Reference(),
+		Id:         id,
+		SnapshotId: snapshotId,
+		Name:       name,
+		Profile:    profile,
+		Crypto:     crypto,
+		Path:       path,
+	}
+
+	res, err := methods.VslmCreateDiskFromSnapshot_Task(ctx, m.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewVslmTask(m.c, res.Returnval), nil
+}
+
+
+func (m *VslmObjectManager) CreateSnapshot(ctx context.Context, id vim.ID, description string) (*VslmTask, error) {
+	req := types.VslmCreateSnapshot_Task{
+		This:        m.Reference(),
+		Id:          id,
+		Description: description,
+	}
+
+	res, err := methods.VslmCreateSnapshot_Task(ctx, m.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewVslmTask(m.c, res.Returnval), nil
+}
+
+func (m *VslmObjectManager) DeleteSnapshot(ctx context.Context, id vim.ID, description string) (*VslmTask, error) {
+	req := types.VslmCreateSnapshot_Task{
+		This:        m.Reference(),
+		Id:          id,
+		Description: description,
+	}
+
+	res, err := methods.VslmCreateSnapshot_Task(ctx, m.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewVslmTask(m.c, res.Returnval), nil
+}
+
+func (m *VslmObjectManager) CloneVStorageObject(ctx context.Context, id vim.ID, spec vim.VslmCloneSpec) (*VslmTask, error) {
```

Comment:
The existing `ObjectManager` calls this wrapper method `Clone`, can we do the same here?  Similar for some of th e methods below, where `VStorageObject` could be dropped from the wrapper method name.